### DISCUSSION TITLE
defaultValue is ignored when using <Controller> with the render prop

### DISCUSSION THREAD

**Discussion by lgibso34** [react-hook-form/react-hook-form#4180] [Q&A] [answered]
Using [this popular codesandbox](https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r?file=/src/index.js) change the defaultValue for "Select" to 10 and you see it works fine when the page reloads.

But if you leave that change and jump into Mui.js and modify the Select component to use the `render` prop instead of `as` it will fail to get the defaultValue of ten that worked fine with `as`.

`<Controller
          render={() => (
            <Select>
              <MenuItem value={10}>Ten</MenuItem>
              <MenuItem value={20}>Twenty</MenuItem>
              <MenuItem value={30}>Thirty</MenuItem>
            </Select>
          )}
          name="Select"
          control={control}
        />`

EDIT (1 hour after post): looks like a solution is to grab `value` from the render prop function and set the `value={value}` on your component. Idk if this is what you guys wanted though

**Comment by nbnsresta** [comment, marked as answer]
A render prop is not just an anonymous function that is called from the Controller. It also contains the props that has been passed which contains the `value` and event callbacks like `onChange` within it. I reckon even the `as` prop operates somewhat similar way to `render`.
In your code sample, you have completely ignored the props that needs to be passed to the Select component.

Just replace your example with 
```
<Controller
  render={(props) => ( 
     <Select {...props}>
        <MenuItem value={10}>Ten</MenuItem>
        <MenuItem value={20}>Twenty</MenuItem>
        <MenuItem value={30}>Thirty</MenuItem>
      </Select>
    )}
   name="Select"
   control={control} />

**Reply by lgibso34** [reply]
Thank you for clarifying.
